tests/mpu_stack_guard/Makefile.ci

```diff
@@ -1,3 +1,4 @@
 BOARD_INSUFFICIENT_MEMORY := \
+    nucleo-f031k6 \
```

Comment:
This board (and stm32f030f4-demo) are already excluded by the `FEATURES_REQUIRED += cortexm_mpu` and should already be marked as unsupported by the CI. Why do you add it here ?

Reply:
Hm, seems like the feature guard doesn't work:
```
[kaspar@ng riot.tmp/tests/mpu_stack_guard (mpu_feature)]$ make info-boards-supported | grep -o nucleo-f031k6
nucleo-f031k6
[kaspar@ng riot.tmp/tests/mpu_stack_guard (mpu_feature)]$ 
```

```
[kaspar@ng riot.tmp/tests/mpu_stack_guard (mpu_feature)]$ BOARD=nucleo-f031k6 make info-debug-variable-CPU_ARCH
cortex-m0
[kaspar@ng riot.tmp/tests/mpu_stack_guard (mpu_feature)]$
```

Seems like CPU_ARCH is not available in Makefile.features.

Reply:
It seems that this PR tries to achieve the same thing as #13236 where CPU_ARCH is moved to Makefile.features. I can cut this part out of #13236 if it also helps here.

Reply:
> I can cut this part out of #13236 if it also helps here.

That would be awesome.

Reply:
Here it is: #13392 :)

Reply:
Thanks a bunch.

```
[kaspar@ng riot.tmp/tests/mpu_stack_guard (mpu_feature)]$ make info-boards-supported | grep -o nucleo-f031k6
[kaspar@ng riot.tmp/tests/mpu_stack_guard (mpu_feature)]$
```